streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -60,8 +60,8 @@ public void init(final InternalProcessorContext context) {
         this.context = context;
         final Serializer<?> contextKeySerializer = ProcessorContextUtils.getKeySerializer(context);
         final Serializer<?> contextValueSerializer = ProcessorContextUtils.getValueSerializer(context);
-        keySerializer = prepareKeySerializer(keySerializer, contextKeySerializer, contextValueSerializer);
-        valSerializer = prepareValueSerializer(valSerializer, contextKeySerializer, contextValueSerializer);
+        keySerializer = prepareKeySerializer(keySerializer, contextKeySerializer, contextValueSerializer, this.name());
+        valSerializer = prepareValueSerializer(valSerializer, contextKeySerializer, contextValueSerializer, this.name());
```

Comment:
It seems better to not pass a potential `null` `contextKeySerializer` or `contextValueSerializer`, but instead forward the `context` directly?

Reply:
Hmm yeah that works too, either way we're doing null checks after pulling the serializer/deserializer through `ProcessorContextUtils` so I don't feel like it changes it drastically. I wonder, is it easier for debugging to have the context serializer/deserializer pulled out here, at the same place we set it eventually? Or is it easier to debug if we pull it out where we're potentially throwing an error, down in `WrappingNullableUtils#prepareSerializer`

Reply:
My argument was mainly about unifying code, ie, try to avoid `null`-checks on different places, but do the `null`-check on a unified place (to avoid that we forget the `null`-check).

Might be good to get the opinion of other on this question.

Reply:
@ableegoldman and I talked about this PR briefly yesterday, maybe she has thoughts here